src/main/java/com/bookJourney/springboot/service/UserService.java

```diff
@@ -44,4 +55,34 @@ public void register(RegistrationRequestDTO registrationRequestDTO) throws UserA
             userRepository.save(user);
         }
     }
+
+    public void processSuccessfulLogin(String username) {
```

Comment:
'process' wskazuje że to tutaj odbywa się cała logika, tak jak wspomniałam w controlerze, możesz przenieść resztę kodu tutaj i zwrócić flagę boolean